nc/source/component/Transform.cpp

```diff
@@ -28,24 +28,48 @@ namespace
 
 namespace nc
 {
-    Transform::Transform(EntityHandle handle, const Vector3& pos, const Quaternion& rot, const Vector3& scale)
+    Transform::Transform(EntityHandle handle, const Vector3& pos, const Quaternion& rot, const Vector3& scale, Transform* parent)
         : ComponentBase(handle),
-          m_matrix{ComposeMatrix(scale, rot, pos)}
+          m_localMatrix{ComposeMatrix(scale, rot, pos)},
+          m_worldMatrix{m_localMatrix},
+          m_parent{parent},
+          m_children{}
     {
         IF_THROW(HasAnyZeroElement(scale), "Transform::Transform - Invalid scale(elements cannot be 0)");
```

Comment:
Might be worthwhile to not call ComposeMatrix in the initializer list, in the case where the IF_THROW would throw, we would not have to perform that matrix multiplication

Reply:
Call it in this scope after the IF_THROW instead